agent/grpc/public/services/peerstream/replication.go

```diff
@@ -240,12 +241,49 @@ func (s *Server) handleUpdateService(
 	sn structs.ServiceName,
 	pbNodes *pbservice.IndexedCheckServiceNodes,
 ) error {
+	_, peering, err := s.GetStore().PeeringRead(nil, state.Query{Value: peerName})
```

Comment:
I think it would improve things to thread the `MutableStreamStatus` all the way through as a parameter to this function. You wouldn't have to access the stream tracker or state store in order to increment/decrement the service count.

One reason is that other bits of the various replication methods will also need access to the stream tracker so threading it through to the fn call here can serve a dual purpose.

Reply:
Added in two params:

* mutable stream status from handle upstream
* logger from handle upstream